wikimedia/wikipedia-ios
# Improve usability of search languages bar (2 of 2)

**Phabricator:** https://phabricator.wikimedia.org/T277058

### Notes
This is part 2 of 2 in improving the usability of the language bar in the search controller ([part 1 here](https://github.com/wikimedia/wikipedia-ios/pull/3908)). This PR focuses on design updates, along with some follow-up work from https://github.com/wikimedia/wikipedia-ios/pull/3908:

1. Adds language code "box" to `SearchLanguageButton`
2. Updates `UIScrollView` to use safe area layout guide to preserve landscape orientation readability
3. Improves RTL support 
4. Scrolls to user's selected language on appearance and when tapped by user 
5. Updates position of `UIScrollView` and more languages button to match design

### Test Steps
Confirm there is behavioral parity with the existing search bar, excepting the changes noted above. Below includes test steps from https://github.com/wikimedia/wikipedia-ios/pull/3908 as well as some additions:

1. Confirm adding and removing languages results in the proper number of language bar buttons displayed.
2. Confirm tapping a search bar button and making a search functions as it did before.
3. Confirm it renders as expected in all themes.
4. Confirm language codes render in square box.
5. Confirm the `UIScrollView` scrolls when the device is in LTR and RTL languages.
6. Confirm the "more" button gradient overlay doesn't obscure the preferred selected language when initially entering the full display search view.

### Screenshots

![SLB_Board](https://user-images.githubusercontent.com/5652327/113939230-91239a80-97b0-11eb-849c-2d9c985484dc.png)

- A. languages with 2, 3, and 4 character codes
- B. alternate themes
- C. 1 includes an in-progress scroll (note scroll indicator clashing with selected underline) 2. additionally includes a [long language](https://en.wikipedia.org/wiki/Taraškievica) code I found ([be-tarask](https://be-tarask.wikipedia.org), unclear – is this code supposed to be this long or not?)
